samples/nrf9160/aws_fota/README.rst

```diff
@@ -31,56 +112,133 @@ Building and running
+   If you already have an nRF Cloud account, you can use the same credentials.
+   Otherwise, create an account and sign in.
+#. In the menu on the left-hand side, select **Firmware**.
+#. Click **Generate new** to create a URL for the firmware image.
+#. Click **Upload** for the URL that you created and select the file ``app_update.bin`` (located in the ``zephyr`` subfolder of your build directory).
+#. Create a job document (a text file) with the following content, replacing *host_url* with the server part of the URL that you created (for example, ``s3.amazonaws.com``) and *file_path* with the path and file name (for example, ``/nordic-firmware-files/0943dfbf-cb10-4eb7-8277-a8b179eaf4ff``):
+   (?should we have an actual file here, or just state ``/nordic-firmware-files/XXXXXXX``?)
+
+   .. parsed-literal::
+      :class: highlight
+
+      {
+        "operation": "app_fw_update",
+        "fwversion": "v1.0.2",
+        "size": 181124,
+        "location": {
+          "protocol": "http:",
+          "host": "*host_url*",
+          "path": "*file_path*"
+         }
+      }
+
+   See `AWS IoT Developer Guide: Jobs`_ for more information about AWS jobs.
+#. Log on to the `AWS S3 console`_.
+#. If you do not have a bucket yet, create one with the default settings.
+#. Select the bucket, click **Upload**, and select your job document.
+   Use the default settings when uploading the file.
+#. Log on to the `AWS IoT console`_, go to **Manage** > **Jobs**, and select **Create a job**.
+#. Click **Create custom job** and enter a unique job ID.
+   Select your device and the job file that you uploaded to AWS S3.
+   Use the default settings for all other options.
+#. Since the sample is configured to subscribe to the ``app_fw_update`` job topic (?what is it actually subscribing?), it picks up the job automatically.
+   This can take several minutes.
+   Select the job in AWS IoT to confirm that it is in progress.
+   Note that it might show as in progress on the overview, while it is actually queued.
+#. In the terminal emulator, observe that the new firmware image is downloaded and installed.
+   You should see output similar to the following::
+
+      (?is there any output worth mentioning?)
+
+#. When the board resets, observe that the sample prints the new application version.
+#. Log on to the `AWS IoT console`_, go to **Manage** > **Things**, and select your thing.
+#. Go to **Shadow** and confirm that the application version has updated.
+
+
+Troubleshooting
+===============
+
+If you host the firmware image on a different server than nRF Cloud, you might get an error message that the content range is not defined.
+(?what is the wording of the error?)
+This is because the sample (?or the download client library?) requires the host server to include the Content-Range field in the HTTP GET header.
+
+To fix this problem, configure the host server to provide this field.
+
+(?should we add anything here about MQTT_Connect -45?)
```

Comment:
It's a common case so I think it's important to say that it's usually caused by incorrectly flashing/copy-pasting/writing certificates to a security tag. Usually by either mixing up the certificate/key types or formatting.

It means that an `operation is not supported on socket` this could be an operation such as using a private key as a public certificate etc.